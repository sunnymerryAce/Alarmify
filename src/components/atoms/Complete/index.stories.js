import React from 'react';
import Complete from './index';

import 'modern-normalize/modern-normalize.css';
import 'swiper/css/swiper.min.css';
import '../../../App.css';

export default {
  title: 'Complete',
};

export const Default = () => <Complete style="block" />;
