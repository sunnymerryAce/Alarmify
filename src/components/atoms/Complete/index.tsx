import React, { forwardRef } from 'react';
import styled from 'styled-components';
import check from 'src/images/baseline-check_circle_outline-24px.svg';

interface Props {
  display: string;
}
const Complete = forwardRef<HTMLDivElement, Props>((props, ref) => {
  const style = {
    display: props.display,
  };
  return (
    <StyledComplete ref={ref} className="Complete" style={style}>
      <img src={check} alt="check" />
    </StyledComplete>
  );
});

export default Complete;

const StyledComplete = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  width: 70px;
  height: 70px;
  img {
    width: 100%;
  }
`;
