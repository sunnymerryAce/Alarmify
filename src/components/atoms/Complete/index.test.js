import React, { createRef } from 'react';
import Complete from './index';
import { shallow } from 'enzyme';

describe('<Complete />', () => {
  const complete = createRef();
  const container = shallow(<Complete ref={complete} display='block'/>);

  test('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('should exist', () => {
    expect(container.exists('.Complete')).toBe(true);
  });

  test('should have proper props', () => {
    expect(container.props()).toMatchObject({
      style: {
        display: 'block'
      },
    });
  });
});
