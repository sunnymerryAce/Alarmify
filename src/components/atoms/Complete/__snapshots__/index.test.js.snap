// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Complete /> should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa dOZbPm Complete\\" style=\\"display:block\\"><img src=\\"baseline-check_circle_outline-24px.svg\\" alt=\\"check\\"/></div>"`;
